doc/tutorial/sam_file/sam_file_writing.cpp

```diff
@@ -0,0 +1,40 @@
+#include <seqan3/test/snippet/create_temporary_snippet_file.hpp>
+// std::filesystem::current_path() / "out.sam" will be deleted after the execution
+create_temporary_snippet_file example_sam{"out.sam", ""};
+
+//![main]
+#include <seqan3/io/sam_file/all.hpp>
+
+using aligned_sequence_type = std::vector<seqan3::gapped<seqan3::dna5>>;
+using alignment_type = std::pair<aligned_sequence_type, aligned_sequence_type>;
+
+using types = seqan3::type_list<std::vector<seqan3::dna5>, std::string, alignment_type>;
+using fields = seqan3::fields<seqan3::field::seq, seqan3::field::id, seqan3::field::alignment>;
```

Comment:
This seems kinda hard, but there isn't another way right now, right?
Did we want to get rid of passing the fields to the file?

This is more prominent in the doc/tutorial/sam_file/sam_file_solution3.cpp example

Reply:
The ultimate goal would be to get rid of the fields, and then we could fix the order of the types.

Reply:
Another way would to re-use the record from an  input file:

```c++
using record_type = decltype(seqan3::sam_file_input{"", seqan3::fields<seqan3::field::seq, seqan3::field::id, seqan3::field::alignment>})::record_type;
```

But the snippet is just for the output, so I thought we should keep both seperate :)

Reply:
Mhm, I think it's ok for now :)